src/MCMCMonitorDataManager/MCMCMonitorData.ts

```diff
@@ -341,7 +371,7 @@ const invalidateStats = <T extends SequenceStatsDict | VariableStatsDict>(statsD
     if (key === undefined) {
         Object.keys(statsDict).forEach(s => statsDict[s].isUpToDate = false)
     }
-    else if (key in Object.keys(statsDict)) {
+    else if (Object.keys(statsDict).includes(key)) {
```

Comment:
First bug: `key in ___` is a Pythonism which does not do what we actually want here.

Reply:
Great find!